MaxText/layers.py

```diff
@@ -704,8 +705,26 @@ def attend(self, query: Array) -> Array:
       in NLP models.
     """
     dtype = self.attend_dtype if self.attend_dtype is not None else self.dtype
-    return jnp.dot(query, jnp.asarray(self.embedding, dtype).T)
```

Comment:
I think it is worthwhile to just fork jnp.dot and add _dot_general parameter to it. That will make the review easier and will be easier to go back to jnp.dot once _dot_general param lands in jax

Reply:
Done.